contracts/utils/math/SafeMath.sol

```diff
@@ -224,4 +224,34 @@ library SafeMath {
             return a % b;
         }
     }
+
+
+
+    /*
+    * @dev
+    *
+    * Returns a number's percentage of another number to a specified decimal place.
+    * The result is to the decimal place of the third parameter passed to the function.
+    * 
+    *
+    * @param:
+    *
+    * uint256 a, uint256 b, uint8 c, string memory errorMessage.
+    * It returns tha `a`% of `b` to `c` decimal places.
+    */
+    function perc(uint256 a, uint256 b, uint8 c, string memory errorMessage) internal pure returns(uint256)
+    {
+        // Require that b is greater than 0 to avoid zero division.
+        require(b > 0, errorMessage);
+        // Check for overflows.
+        require((a * b * (10 ** c)) < type(uint256).max, errorMessage);
```

Comment:
this check doesn't really make sens. 

If `(a * b * (10 ** c)) > type(uint256).max`, the multiplication will revert because of solidity 0.8.x overflow protection... So the only case where this would revert is not equality ... why would you have a specific revert is equality?

Reply:
Thank you for your reply.

Reply:
well, if there is no overflow protection at the solidity level, and overflow happens, this test will not catch it.